test/common/wpt.js

```diff
@@ -6,9 +6,92 @@ const fs = require('fs');
 const fsPromises = fs.promises;
 const path = require('path');
 const events = require('events');
+const os = require('os');
 const { inspect } = require('util');
 const { Worker } = require('worker_threads');
 
+function getBrowserProperties() {
+  const release = !!process.release.sourceUrl;
+  const browser = {
+    browser_channel: release ? 'stable' : 'experimental',
+    browser_version: process.version,
+  };
+
+  if (release) {
+    browser.browser_version = process.version;
+  } else if (process.env.NODE_REVISION) {
+    browser.browser_version = process.version.replace('pre', process.env.NODE_REVISION);
+  }
+
+  return browser;
+}
+
+function getOs() {
+  switch (os.type()) {
+    case 'Linux':
+      return 'linux';
+    case 'Darwin':
+      return 'mac';
+    case 'Windows_NT':
+      return 'win';
+    default:
+      return 'unknown';
+  }
+}
+
+class WPTReport {
+  constructor() {
+    this.results = [];
+    this.time_start = Date.now();
+  }
+
+  addResult(name, status) {
+    const result = {
+      test: name,
+      status,
+      subtests: [],
+      addSubtest(name, status, message) {
+        const subtest = {
+          status,
+          name,
+        };
+        if (message) subtest.message = message;
+        this.subtests.push(subtest);
+        return subtest;
+      },
+    };
+    this.results.push(result);
+    return result;
+  }
+
+  write() {
+    this.time_end = Date.now();
+    this.results = this.results.filter((result) => {
+      return result.status === 'SKIP' || result.subtests.length !== 0;
+    }).map((result) => {
+      result.test = result.test.replace(/\.js(?:\?|$)/, '.html');
```

Comment:
The filename of the test but keeping optional variant `?1-1000` and renaming to `.html` to align with regular WPT runner convention.